lang/src/encode_input_data.rs

```diff
@@ -0,0 +1,82 @@
+use ink_core::memory::vec::Vec;
```

Comment:
please just remove this entire file. it is confusing and doesn't even test the calling infrastructure. we will find suitable tests for the call infrastructure at a later point.